src/Symfony/Component/Notifier/Bridge/Smsapi/Tests/SmsapiTransportTest.php

```diff
@@ -44,4 +46,52 @@ public function unsupportedMessagesProvider(): iterable
         yield [new ChatMessage('Hello!')];
         yield [$this->createMock(MessageInterface::class)];
     }
+
+    public function createClient(int $statusCode, array $content): HttpClientInterface
+    {
+        $response = $this->createMock(ResponseInterface::class);
+        $response
+            ->method('toArray')
+            ->willReturn($content);
+
+        $response
+            ->method('getStatusCode')
+            ->willReturn($statusCode);
+
+        $client = $this->createMock(HttpClientInterface::class);
+        $client
+            ->method('request')
+            ->willReturn($response);
+
+        return $client;
+    }
+
+    public function responseProvider(): iterable
+    {
+        $responses = [
+            ['status' => 200, 'content' => '{"error":101,"message":"Authorization failed"}', 'errorMessage' => 'Unable to send the SMS: "Authorization failed".'],
+            ['status' => 500, 'content' => '{}', 'errorMessage' => 'Unable to send the SMS: "unknown error".'],
```

Comment:
can you add a case where the response is not json

Reply:
done